lab/lablib/tone_helpers/schedule_music.js

```diff
@@ -0,0 +1,64 @@
+import { Cycle, Gap, Loop, Parallel, Sequential } from "../music/Timeline.js";
+import { Rational } from "../Rational.js";
+import { ToneClip } from "./compile_music.js";
+import { to_tone_time } from "./measure_notation.js";
+
+/**
+ * 
+ * @param {Rational} offset 
+ * @param {Loop<ToneClip>} loop 
+ */
+export function schedule_loop(offset, loop) {
+    const { duration, child } = loop;
+    if (duration < child.duration) {
+        throw new Error("not implemented: crop loop")
+    } else if (duration > child.duration) {
+        throw new Error("not implemented:")
+    }
+
+    schedule_clips(offset, child);
+}
+
+/**
+ * 
+ * @param {Rational} offset 
+ * @param {Cycle<ToneClip>} clips 
+ */
+function schedule_cycle(offset, clips) {
+    throw new Error("Function not implemented.");
+}
+
+/**
+ * 
+ * @param {Rational} offset 
+ * @param {import("../music/Timeline").Timeline<ToneClip>} clips 
+ */
+export function schedule_clips(offset, clips) {
```

Comment:
I need a way to decouple this for unit testing. And also it might be handy to have a list of the clips.

Perhaps making this a generic `Timeline<T>` would allow me to use a stub instead of ToneClip for unit tests...